core/backend/src/SqliteStatement.ts

```diff
@@ -11,21 +11,23 @@ import { ECJsNames, IModelError } from "@bentley/imodeljs-common";
 import { IModelJsNative } from "@bentley/imodeljs-native";
 import { IModelHost } from "./IModelHost";
 
-/** Marks a string as either an [Id64String]($bentleyjs-core) or [GuidString]($bentleyjs-core), so
- * that it can be passed to the [bindValue]($backend.SqliteStatement) or [bindValues]($backend.SqliteStatement)
- * methods of [SqliteStatement]($backend).
- * @internal
+/** an object returned by [[SqliteStatement.getRow]]
+ * @public
  */
-export interface StringParam {
-  id?: Id64String;
-  guid?: GuidString;
+export interface RowValue {
+  [propName: string]: Uint8Array | number | string | undefined;
 }
 
 /** parameter Index (1-based), or name of the parameter (including the initial ':', '@' or '$')
  * @public
  */
 export type BindParameter = number | string;
 
+/** A value that can be bound with [[SqliteStatement.bindValue]] or [[SqliteStatement.bindValues]]
+ * @public
+ */
+export type ParameterValue = undefined | null | number | boolean | string | Uint8Array | { id?: Id64String, guid?: GuidString };

```

Comment:
Following would more accurately enforce that exactly one of id or guid is permitted.
`| { id: Id64String } | { guid: GuidString }`
Weird to add `null` here where we discourage its use everywhere else and it behaves exactly the same as `undefined`? Or trying to match SQL concept of NULL?

Reply:
null is accepted in `bindValue`, so I added it. But, probably a good idea to give compile errors so it isn't propagated.

Reply:
i removed null. @khanaffan was using it in `NativeAppStorage`, but i don't think that was a good idea.